### PR TITLE
feat: add hotpath for profiling

### PR DESCRIPTION
**Pull request by 0xEgao**
# Pull Request

## Description
Adds hotpath crate for profiling taker and maker.
This feature is entirely opt-in — nothing runs unless explicitly compiled with --features "hotpath, hotpath-alloc" and pass --hotpath on the CLI.

## Related Issue(s)
Closes #

## Type of Change
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Code refactor / performance improvement
- [ ] Documentation update only
- [ ] CI / Docker / Build changes
- [ ] Other (please describe):

## Protocol Version(s) Affected
- [ ] Legacy (ECDSA — `messages.rs`, `contract.rs`, `handlers.rs`)
- [ ] Taproot-Musig2 (`messages2.rs`, `contract2.rs`, `handlers2.rs`)
- [ ] Both
- [x] Neither (infrastructure / tooling only)

> **Note:** When modifying protocol logic, changes must be made to **both** versions unless the
> scope is explicitly version-specific.

## Affected Component(s)
- [ ] **makerd** (background daemon)
- [ ] **taker** (CLI client)
- [ ] **maker-cli** (command-line tool)
- [ ] Core protocol / cryptography / library
- [ ] **watch_tower** (contract monitoring & recovery)
- [ ] Docker / deployment scripts
- [ ] Tests / test framework
- [ ] Documentation (`docs/`)
- [ ] Other (please specify):

### Code Quality
- [x] I ran `cargo +nightly fmt --all` and committed the result
- [x] I ran `cargo +stable clippy --all-features --lib --bins --tests -- -D warnings` with zero warnings
- [x] I ran `cargo +stable clippy --examples -- -D warnings` with zero warnings
- [x] I ran `RUSTDOCFLAGS="-D warnings" cargo +nightly doc --all-features --document-private-items --no-deps` with zero warnings
- [ ] Pre-commit git hook passes (`ln -s ../../git_hooks/pre-commit .git/hooks/pre-commit` if not already set)

### Testing
- [ ] All unit tests pass (`cargo test`)
- [ ] Integration tests pass (`cargo test --features integration-test`)
- [ ] Changes were manually tested on **regtest**
- [ ] End-to-end maker ↔ taker swap tested (where applicable)
- [ ] Test-only code is gated behind `#[cfg(feature = "integration-test")]`

### Documentation
- [x] Relevant files in the `docs/` folder were updated
- [ ] Complex logic is commented

### Security & Privacy (Critical)
- [ ] This change preserves **trustlessness** and **atomic swap guarantees**
- [ ] No regression in **sybil resistance** or **fidelity bond** logic
- [ ] Tor anonymity and P2P message flow were reviewed
- [ ] No new attack vectors or trust assumptions introduced
- [ ] Edge cases and error handling considered
- [ ] ZMQ subscription integrity (raw tx/block feed) is unaffected
- [ ] `integration-test` feature flag is not reachable in production code paths

## Summary by CodeRabbit

* **New Features**
  * Optional Hotpath performance profiling via Cargo features; usable for maker and taker binaries with a new --hotpath CLI flag
  * Profiling produces pretty-formatted JSON reports under <DATA_DIR>/hotpath/ and prints summary tables (timing and optional allocation) to the console after swaps

* **Chores**
  * Added Hotpath dependency and feature flags to enable conditional compilation

* **Documentation**
  * Added profiling guide with usage examples and report info

### CODE REVIEW

File: src/protocol/musig2.rs [hunk 1]
```diff
@@ -13,6 +13,7 @@ use secp256k1::{
 use crate::protocol::error::ProtocolError;
 
 /// get aggregated public key from two public keys
+#[hotpath::measure]
 pub fn get_aggregated_pubkey(pubkey1: &PublicKey, pubkey2: &PublicKey) -> XOnlyPublicKey {
```
[the remark this answers is not in the document]
Comment (0xEgao) [reply src/protocol/musig2.rs:17]: All `#[hotpath::measure]` macros are no-op until the binary is build using hotpath feature.

**Review by mojoX911 — APPROVED** [review]
Ack
